### ISSUE TITLE
High Available Kubernetes Problems (ETCD) and http_proxy

### ISSUE THREAD

**Issue by devopsop** [kubernetes/kubernetes#61907]
Greeeeetings,

I am currently following this HA Guide from Kubernetes's official Website: 
https://kubernetes.io/docs/setup/independent/high-availability/

I have to be behind an http_proxy so it makes my life super hard!

Alright so I want to have 3 masters:
Master1 - Master2 - Master3

I started by installing Docker (with http proxy variables in order for it to work properly) and installed kubernetes 1.9.6 on all 3 machines (kubeadm – kubectl and kubelet). (disabled firewalld / swap / selinux)

Next, I downloaded cfssl and cfssljson on all 3 machines.

On Master1:
- I created ca-config.json ca-csr.json client.json
- Generated the ca and client certs with the commands on the guide

Copied the .pem on to Master 2 and Master 3

Now on all 3 masters I did the followings:
-	cfssl print-defaults csr > config.json
-	Change the ip and hostname in it
-	Generate the peer and server certs
-	Installed and created the service file for etcd
-	Started etcd

There is no errors nothing in the journalctl logs or status. Everything is green OK!

Next I kubeadm init using no_proxy and the appropriate config.yml file on Master1

After it is completed, I look at the etcd journalctl logs on etcd and I get this every 10 seconds:
```Mar 29 13:49:18 infra0 etcd[1769]: rejected connection from "Master1_IP:51774" (error "EOF", ServerName "")```

When I copy the pki folder and try to init Master2 I get the same error log but it won’t complete It stays stuck @ the pulling image message....

I am stuck..... I've hit a wall....

Is there a better way to do it? the etcd cluster seems to be the issue here! Would it be because of my http_proxy? Thanks

**Comment by devopsop** [comment]
I have just tested again locally using vagrant and without a http_proxy and get the exact same error.

One more thing whenever I curl http or https on 2379 i get this:
![image](https://user-images.githubusercontent.com/36001380/38108515-168190c0-3363-11e8-84c5-65d5a12cadee.png)

**Comment by devopsop** [comment]
Fixed by downgrading etcd from 3.3 to 3.2

But I still cant init Master2 and Master3

error from kubelet: 
```
Failed to list *v1.Pod: Get https://192.168.50.20:6443/api/v1/pods?fieldSelector=spec.nodeName%3Dinfra1&limit=500&resourceVersion=0: x509: certificate is valid for 10.96.0.1, 192.168.50.10, 192.168.50.10, not 192.168.50.20
```

**Comment by devopsop** [comment]
Problem solved, I had copied everything from Master0. Instead, I had to copy everything except the api-server.key and .crt
